### PR TITLE
Bumps gtest to 20210126 and adds QNX 7.1 support

### PR DESCRIPTION
**Pull request by mjvankampen**
Specify library name and version:  **gtest/cci-26012021**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/gtest/all/conandata.yml
Comment (Croydon) [inline recipes/gtest/all/conandata.yml:8]: This has to be YYYY-MM-DD not the other way around

File: recipes/gtest/all/conandata.yml
Comment (Croydon) [inline recipes/gtest/all/conandata.yml:20]: ^

File: recipes/gtest/config.yml
Comment (Croydon) [inline recipes/gtest/config.yml:6]: ^

File: recipes/gtest/all/conandata.yml
Comment (mjvankampen) [reply recipes/gtest/all/conandata.yml]: Duh, well that's embarrassing. Fixed.

File: recipes/gtest/all/conandata.yml
Comment (Croydon) [inline recipes/gtest/all/conandata.yml:8]: Sorry that I did not notice it the first time. But it has to be a dot instead of a dash. See also https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#what-version-should-packages-use-for-libraries-without-official-releases

`cci.20210126`

Comment (mjvankampen) [reply recipes/gtest/all/conandata.yml]: No worries, I should read better :).

**Comment by jgsogo** [comment]
Same [here](https://github.com/conan-io/conan-center-index/pull/4398#issuecomment-768891921). CI has failed with _"Unable to retrieve commit message"_.

**Comment by Croydon** [comment]
@jgsogo Just as a note, this time around this might be actually a GitHub problem. [Bincrafters-conventions](https://github.com/bincrafters/bincrafters-conventions/actions) started to have sporadically issues with git clones. It claims that the repositories do not exist, while the repo list is from the GitHub API..

**Comment by mjvankampen** [comment]
@jgsogo @Croydon is this still an issue? Maybe worth to re-run?

**Comment by Croydon** [comment]
You can restart the CI by closing this PR for ~10 seconds and then reopening it

**Review by jgsogo — DISMISSED** [review]
I see `gtest` has moved to live-at-head approach, so we need to start creating `cci.<date>` versions.

File: recipes/gtest/all/conanfile.py
Comment (SpaceIm) [inline recipes/gtest/all/conanfile.py:39]: Are you sure? It worked in 1.10.0 without these checks. Visual Studio 2017 seems excessive for C++11, gcc 5 and clang 5 (clang 3.4 has wide support of C++14).

File: recipes/gtest/all/conanfile.py
Comment (SpaceIm) [inline recipes/gtest/all/conanfile.py:32]: Why 13 now? It was 12 before this modification.

File: recipes/gtest/all/conanfile.py
Comment (mjvankampen) [reply recipes/gtest/all/conanfile.py:39]: These are the compilers officially supported by gtest. Note that only a warning is given, not an error.

Ref. https://github.com/google/googletest#compilers

File: recipes/gtest/all/conanfile.py
Comment (mjvankampen) [reply recipes/gtest/all/conanfile.py:32]: From officially supported list: https://github.com/google/googletest/tree/release-1.8.1#windows-requirements only 2015 and up are supported. 2013 is 12.

Comment (SpaceIm) [reply recipes/gtest/all/conanfile.py:32]: Visual Studio 2015 = Visual Studio 14
Visual Studio 2013 = Visual Studio 12

File: recipes/gtest/all/conanfile.py
Comment (SpaceIm) [reply recipes/gtest/all/conanfile.py:39]: Visual Studio 2015 = Visual Studio 14

File: recipes/gtest/all/conanfile.py
Comment (mjvankampen) [reply recipes/gtest/all/conanfile.py:32]: Hmmm both should be 14 then according to google as 1.8.1, 1.10.0 and head require 2015. Thanks, fixed.

File: recipes/gtest/all/conanfile.py
Comment (mjvankampen) [reply recipes/gtest/all/conanfile.py:39]: Fixed, I'm not sure what we should do with 1.10.0 as it did work I guess, but according to the maintainer it should not...

File: recipes/gtest/all/conanfile.py
Comment (prince-chrismc) [inline recipes/gtest/all/conanfile.py:112]: Only this exact version requires a system lib?

File: recipes/gtest/all/conanfile.py
Comment (prince-chrismc) [inline recipes/gtest/all/conanfile.py:67]:
```suggestion
        extracted_dir = glob.glob("googletest-*/")[0]
```

File: recipes/gtest/all/conanfile.py
Comment (mjvankampen) [reply recipes/gtest/all/conanfile.py:112]: Since 7.1 regex was put in a separate library. Maybe also for next versions but there was over 3 years between 7.0 and 7.1 so I just put it at this one as there is no telling...

File: recipes/gtest/all/conanfile.py
Comment (mjvankampen) [reply recipes/gtest/all/conanfile.py:67]: Duh, thanks!

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by intelligide — APPROVED** [review]

**Comment by permoberg-vs** [comment]
@mjvankampen I think this broke support for 1.10.0 with GCC 4.8.5 (default in centos7). 1.10.0 has been working fine for us before this was merged.

Why did this go into an update of 1.10.0 and not into 1.10.1?

Can this be reverted or updated so that we can use 1.10.0 with GCC 4.8.5 again?

**Comment by prince-chrismc** [comment]
there does not seem to be any changes applied to `1.10.0` https://github.com/conan-io/conan-center-index/pull/4380/files#diff-5c100d496cc9af24a1d8cdef140c0889469225242b048ccb1b79f1e242546290R20-R22

Is it an ABI problem? https://github.com/conan-io/conan-center-index/pull/4380/files#diff-30c143c486109e8ef87666fdfc05fe43889e15121afa1addd541aa409c91ee95R37 might be it?

**Comment by mjvankampen** [comment]
Hi @permoberg-vs It seems I mixed up the requirements for 1.10.0 the requirement is a c++11 compiler, nothing else. For newer version gcc 5 is required. If you make a bug I will fix it.

**Comment by permoberg-vs** [comment]
> Hi @permoberg-vs It seems I mixed up the requirements for 1.10.0 the requirement is a c++11 compiler, nothing else. For newer version gcc 5 is required. If you make a bug I will fix it.

Hi @mjvankampen, that would be awesome if it could be fixed, I've created this issue for it:
https://github.com/conan-io/conan-center-index/issues/4659

**Pull request merged**
